extension/extension.go

```diff
@@ -0,0 +1,191 @@
+package extension
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"net/http"
+	"os"
+	"os/exec"
+	"path"
+	"runtime"
+	"strings"
+	"time"
+
+	"github.com/google/go-github/github"
+)
+
+const defaultRepoPrefix = "optimus-extension-"
+
+// Extension is manager for extension
+type Extension struct {
+	ctx context.Context
+
+	githubClient *github.Client
+	httpClient   *http.Client
+
+	manifest *Manifest
+
+	reservedCommands []string
+}
+
+// NewExtension initializes extension
+func NewExtension(
+	ctx context.Context, manifest *Manifest,
+	githubClient *github.Client, httpClient *http.Client,
+	reservedCommands ...string,
+) (*Extension, error) {
+	return &Extension{
+		ctx:              ctx,
```

Comment:
Why do we have `context` as part of the member variable?

Reply:
The context is required when trying to get the latest release information. I agree that we can create a new context inside the calling method. But, the context is also used when trying to download the latest release binary file. So, rather than re-initializing it, perhaps we can re-use it by sending it through the argument.

Reply:
I think it's better if we move it to `Install` as the first argument so the caller can decide to inject the context.

Reply:
I have no strong opinion whether it is better in the initializer or in the method since I saw some repos use it in either of them. Sure, I can move it `Install`.